zk_ee/src/system/mod.rs

```diff
@@ -46,33 +44,26 @@ pub trait SystemTypes {
     /// Handles all side effects and information from the outside world.
     type IO: IOSubsystem<IOTypes = Self::IOTypes, Resources = Self::Resources>;
 
-    type Memory: MemorySubsystem<Allocator = Self::Allocator>;
-
     /// Common system functions implementation(ecrecover, keccak256, ecadd, etc).
     type SystemFunctions: SystemFunctions<Self::Resources>;
 
     type Logger: Logger + Default;
 
     // These are just shorthands. They are completely defined by the above types.
     type IOTypes: SystemIOTypesConfig;
-    type Resources: Resources;
+    type Resources: Resources + Default;
     type Allocator: Allocator + Clone + Default;
 }
 pub trait EthereumLikeTypes: SystemTypes<IOTypes = EthereumIOTypesConfig> {}
 
 pub struct System<S: SystemTypes> {
```

Comment:
I think at this point we need only one System trait which will contain IO + metadata method

Reply:
We can do it in separate PR.

Actually, it's how it was at the very beginning, before IO and memory split. We actually split it to allow access to memory and io at the same time, but it's not relevant anymore.